### DISCUSSION TITLE
useFetch error type

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#16798] [Questions]
my api is returning an error using CreateError
I'm trying to correctly catch the error on the client and display a message
```ts
	const onSubmit = async () => {
		const { error } = await useFetch<IServerResponse>('/api/auth/login', {
			method: 'POST',
			body: {
				email: email.value,
				password: pass.value,
			},
		});
		if (error.value) {
			console.log(error.value.data.statusMessage); // here Typescript error of unknown 'data' feild
			return;
		}
		await useRouter().replace('/');
	};
```
how can I make typescript understand the error correctly?

**Comment by sevillaarvin** [comment]
According to this:

https://github.com/nuxt/framework/blob/c98e5c774d8ef3eb1fb5c3642df4753a7585dbed/packages/nuxt/src/app/composables/asyncData.ts#L44

`error` has type `ErrorT`

https://github.com/nuxt/framework/blob/c98e5c774d8ef3eb1fb5c3642df4753a7585dbed/packages/nuxt/src/app/composables/fetch.ts#L19

which is just `FetchError`

https://github.com/unjs/ohmyfetch/blob/a5ebc970414a46f403f552812a46850adcbc71f7/src/error.ts#L7

which has a property `data`

So perhaps it should be `error.data.statusMessage`?
